Lesson17/elifcetin/script.js

```diff
@@ -0,0 +1,619 @@
+}
+getDayOfWeek(2)
+getDayOfWeek(6)
+
+// value == value
+/// value and type === value and type
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+function repeatWord(word, times){
+  if ( typeof word !=='string' || typeof times !== 'number'){
+  }
+   console.log(word.repeat(times))
+}
+repeatWord("hii", 6)
+
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+function censorWord(sentence, target){
+   if (typeof sentence !== 'string' || typeof target !== 'string') {
+    return "invalid"
+  }
+    const censored = sentence.replaceAll(target, "****");
+  console.log(censored);
+}
+censorWord("hello my name is ali, ali", "ali"); 
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/ 
+
+function startsWithA(str) {
+  if (typeof str !== 'string'){
+    console.log('please provide string')
+    return
+  } 
+  
+   if(str.charAt(0) === 'A') {
+    console.log("Starts with A")
+  } else{
+    console.log("Does not start with A")
+  }
+}
+startsWithA("A very beautiful day")
+
+
+
+
+
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+function sliceLastN(text, n){
+   const result = text.slice(-n)
+   console.log(result)
+  }
+  sliceLastN("hello", 2)
+
+
+
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+function gradeChecker(score){
+  if(score >= 90){
+    console.log("A")
+  } else if(score >= 80){
+    console.log("B")
+  } else if(score >= 70){
+    console.log("C")
+  } else if(score >= 60){
+    console.log("D")
+  } else{
+    console.log("F")
+  }
+}
+gradeChecker(85)
+
+
+
+/*
+18. Character Replacement
+   - Define a function `replaceCharacter(str, oldChar, newChar)` that uses .replaceAll()
+     (or a loop) to swap all occurrences of oldChar with newChar.
+   - Log the result.
+*/
+function replaceCharacter(str, oldChar, newChar){
+   const result = str.replaceAll(oldChar, newChar)
+   console.log(result);
+}
+replaceCharacter("apple", "l", "p")
+
+/*
+19. Title Case a Sentence
+   - Define a function `titleCase(sentence)` that:
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word
+     - Joins them back
+   - Log the transformed string.
+*/
+function titleCase(sentence){
+  const words = sentence.split(" ")
+   const capitalized = words.map(word =>
```

Comment:
For every word in the words array, we take the first letter and convert it to uppercase, then combine it with the rest of the word in lower case. All these modified words will be then returned in a form of array.